src/units/zone_signer.rs

```diff
@@ -934,6 +952,17 @@ fn collect_zone(zone: Zone) -> Vec<StoredRecord> {
     zone.read()
         .walk(Box::new(move |owner, rrset, _at_zone_cut| {
             let mut unlocked_records = passed_records.lock().unwrap();
+
+            // SKIP DNSSEC records that should be generated by the signing
+            // process (these will be present if re-signing a published signed
+            // zone rather than signing an unsigned zone)
+            if matches!(
```

Comment:
This assumes we are the ones adding them to the zone via `dnst keyset`.

Should we also strip CDS and CDNSKEY here? Or should we actually always be stripping them (and other records) from the zone in the signer irrespective of whether the zone was previously signed or not?

Reply:
Yes, always stripping DNSKEY/CDS/CDNSKEY is the right thing to do. In the future keyset will have an apex_remove field with a list of Rtypes to remove from the apex.